library/std/src/path.rs

```diff
@@ -2405,6 +2418,18 @@ impl Path {
     ///
     /// This is a convenience function that coerces errors to false. If you want to
     /// check errors, call [`fs::metadata`].
+    ///
+    /// An example of proper error handling.
+    /// ```no_run
+    /// use std::path::Path;
+    /// use std::io::ErrorKind;
+    /// use std::fs::metadata;
+    /// let exists = metadata(Path::new("does_not_exist.txt"))
+    ///     .map(|_| true)
+    ///     .or_else(|error| if error.kind() == ErrorKind::NotFound { Ok(false) } else { Err(error) } )
+    ///     .expect("failed to check existence of a file");
+    /// assert!(!exists);
```

Comment:
I think it would be clearer with the 3 cases separated as such, rather than having to reason through these not-a-great-fit combinators.

```rust
use std::fs;
use std::io::ErrorKind;

let exists = match fs::metadata("does_not_exist.txt") {
    Ok(_) => true,
    Err(err) if err.kind() == ErrorKind::NotFound => false,
    Err(err) => panic!("failed to check existence of file: {}", err),
};

assert!(!exists);
```

Reply:
Separately, it strikes me as odd to refer to a panic (`expect`) as "proper error handling". Could this be reworded?

Reply:
I agree, same change was made in the new `try_exists()` method. I'm not sure if/how the documentation should be changed taking into account there's a method in nightly. I guess referring to it from stable is not usual, right?

I assumed panic is OK in this context because of the `assert!` below, but changing to `return Err(err)` would make sense.